pkg/core/xds/merge.go

```diff
@@ -0,0 +1,112 @@
+package xds
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+
+	jsonpatch "github.com/evanphx/json-patch/v5"
+)
+
+// indicates that all slices that start with this prefix will be appended, not replaced
+const appendSlicesPrefix = "Append"
+
+func MergeConfs(confs []interface{}) (interface{}, error) {
+	if len(confs) == 0 {
+		return nil, nil
+	}
+
+	resultBytes := []byte{}
+	for _, conf := range confs {
+		confBytes, err := json.Marshal(conf)
+		if err != nil {
+			return nil, err
+		}
+		if len(resultBytes) == 0 {
+			resultBytes = confBytes
+			continue
+		}
+		resultBytes, err = jsonpatch.MergePatch(resultBytes, confBytes)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	result, err := newConf(reflect.TypeOf(confs[0]))
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(resultBytes, result); err != nil {
+		return nil, err
+	}
+
+	clearAppendSlices(reflect.ValueOf(result)) // clear appendable slices, so we won't duplicate values of the last conf
```

Comment:
nit: can we move the comment to the line above? I think it's just a bit easier to read:
```go
// clear appendable slices, so we won't duplicate values of the last conf
clearAppendSlices(reflect.ValueOf(result))
```
but I don't have strong opinion on that

Reply:
I don't mind